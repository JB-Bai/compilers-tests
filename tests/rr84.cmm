int h2MCROk(){
	015;
	while(2679.37) 615.12e+2.QOSeWLglLn || 1.629;
	return P64Ff2_p[PoUXKl9(OLY6XgF8, 0x3e8ae,(- 53.89 = h3jEtyHb(84.47)) !=(iemjx2fr()), YtulaCvwsSj + 32.48e4 = 0xa1[061.666E3 * 687441 + - jyXt1lbcy1 && W5jdaw4PX] || 1473, oWkgLP() / 13.9344,(raxauQZ || SNzoM_B(UwEi8MT, ! 857.955E4, 639.061.M00V, 0x2A) * 5283 /(OPqvHPa[e0Fw8R() - 59858 || b4l6Q0qz().ItrBmL].vqeV != zEiXJ + Dk0G9FzL - 814.789 < HDHvfVsPAWP || 0x232 = 0243 || 95.9511) * 0xcB) + 0xa8 || 83.94e+2 || - tcFUA9 <=(0x3) = 0x2e10C = 022 * ! gHnDpr2(qouWvU9, PLRI, aJvTKi(E0o02ZA, 1105.396e2 + iCAPvAM1[0xdA3 = 141.1639] / 6.774 || hxSRnq5 * 07452 && 0136 / Sf6ZMNjMM > h67LRgL / - 2373 > 0x9d6C > 823.55 + ! _raTa() <=(917.10 - 0xBF5) || ! - 88 / iLtmQ2tCy || Laid2RnZ.AhrAlj =(OdQPWvGa) && 4285, NSyKr, 00233, ru88E1v()),(Cbv_Bq8Jc) - ! CHTt8HaK[JcC3()] -(sXkry7ail()[N2Uz0L])))];
}
float qohNMKpB3S, fm0DU_Qh[0xbcd5], B6PlP0S0[04114], jhwWoRt4, n2phZCzbp;
struct _OxPpQQ{
	int zyzBQk, mqbddYh;
	int FAGTK, q3X6coKfm = 63014, FB1rbf = 1718.28e9;
}
JfaFtGL(float HkEQoZPrV, int CWdOQ[0xb39c]){
	while(0xd3) if(5.924[htNKoAV]) 0902.15;
	NTe0jU6 || UUXOBoap2y - 187.527e-4.uqBo7A_;
}
struct FNtTI_Kfnh;
int iK1NbTFcE;
int GN13DP[0x9];

