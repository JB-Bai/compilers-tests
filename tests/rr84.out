Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: h2MCROk
        LP
        RP
      CompSt (1)
        LC
        StmtList (2)
          Stmt (2)
            Exp (2)
              INT: 13
            SEMI
          StmtList (3)
            Stmt (3)
              WHILE
              LP
              Exp (3)
                FLOAT: 2679.370117
              RP
              Stmt (3)
                Exp (3)
                  Exp (3)
                    Exp (3)
                      FLOAT: 61512.000000
                    DOT
                    ID: QOSeWLglLn
                  OR
                  Exp (3)
                    FLOAT: 1.629000
                SEMI
            StmtList (4)
              Stmt (4)
                RETURN
                Exp (4)
                  Exp (4)
                    ID: P64Ff2_p
                  LB
                  Exp (4)
                    ID: PoUXKl9
                    LP
                    Args (4)
                      Exp (4)
                        ID: OLY6XgF8
                      COMMA
                      Args (4)
                        Exp (4)
                          INT: 256174
                        COMMA
                        Args (4)
                          Exp (4)
                            Exp (4)
                              LP
                              Exp (4)
                                Exp (4)
                                  MINUS
                                  Exp (4)
                                    FLOAT: 53.889999
                                ASSIGNOP
                                Exp (4)
                                  ID: h3jEtyHb
                                  LP
                                  Args (4)
                                    Exp (4)
                                      FLOAT: 84.470001
                                  RP
                              RP
                            RELOP
                            Exp (4)
                              LP
                              Exp (4)
                                ID: iemjx2fr
                                LP
                                RP
                              RP
                          COMMA
                          Args (4)
                            Exp (4)
                              Exp (4)
                                Exp (4)
                                  ID: YtulaCvwsSj
                                PLUS
                                Exp (4)
                                  FLOAT: 324800.000000
                              ASSIGNOP
                              Exp (4)
                                Exp (4)
                                  Exp (4)
                                    INT: 161
                                  LB
                                  Exp (4)
                                    Exp (4)
                                      Exp (4)
                                        Exp (4)
                                          FLOAT: 61666.000000
                                        STAR
                                        Exp (4)
                                          INT: 687441
                                      PLUS
                                      Exp (4)
                                        MINUS
                                        Exp (4)
                                          ID: jyXt1lbcy1
                                    AND
                                    Exp (4)
                                      ID: W5jdaw4PX
                                  RB
                                OR
                                Exp (4)
                                  INT: 1473
                            COMMA
                            Args (4)
                              Exp (4)
                                Exp (4)
                                  ID: oWkgLP
                                  LP
                                  RP
                                DIV
                                Exp (4)
                                  FLOAT: 13.934400
                              COMMA
                              Args (4)
                                Exp (4)
                                  Exp (4)
                                    Exp (4)
                                      Exp (4)
                                        Exp (4)
                                          LP
                                          Exp (4)
                                            Exp (4)
                                              ID: raxauQZ
                                            OR
                                            Exp (4)
                                              Exp (4)
                                                Exp (4)
                                                  Exp (4)
                                                    ID: SNzoM_B
                                                    LP
                                                    Args (4)
                                                      Exp (4)
                                                        ID: UwEi8MT
                                                      COMMA
                                                      Args (4)
                                                        Exp (4)
                                                          NOT
                                                          Exp (4)
                                                            FLOAT: 8579550.000000
                                                        COMMA
                                                        Args (4)
                                                          Exp (4)
                                                            Exp (4)
                                                              FLOAT: 639.060974
                                                            DOT
                                                            ID: M00V
                                                          COMMA
                                                          Args (4)
                                                            Exp (4)
                                                              INT: 42
                                                    RP
                                                  STAR
                                                  Exp (4)
                                                    INT: 5283
                                                DIV
                                                Exp (4)
                                                  LP
                                                  Exp (4)
                                                    Exp (4)
                                                      Exp (4)
                                                        Exp (4)
                                                          Exp (4)
                                                            Exp (4)
                                                              Exp (4)
                                                                ID: OPqvHPa
                                                              LB
                                                              Exp (4)
                                                                Exp (4)
                                                                  Exp (4)
                                                                    ID: e0Fw8R
                                                                    LP
                                                                    RP
                                                                  MINUS
                                                                  Exp (4)
                                                                    INT: 59858
                                                                OR
                                                                Exp (4)
                                                                  Exp (4)
                                                                    ID: b4l6Q0qz
                                                                    LP
                                                                    RP
                                                                  DOT
                                                                  ID: ItrBmL
                                                              RB
                                                            DOT
                                                            ID: vqeV
                                                          RELOP
                                                          Exp (4)
                                                            Exp (4)
                                                              Exp (4)
                                                                ID: zEiXJ
                                                              PLUS
                                                              Exp (4)
                                                                ID: Dk0G9FzL
                                                            MINUS
                                                            Exp (4)
                                                              FLOAT: 814.789001
                                                        RELOP
                                                        Exp (4)
                                                          ID: HDHvfVsPAWP
                                                      OR
                                                      Exp (4)
                                                        INT: 562
                                                    ASSIGNOP
                                                    Exp (4)
                                                      Exp (4)
                                                        INT: 163
                                                      OR
                                                      Exp (4)
                                                        FLOAT: 95.951103
                                                  RP
                                              STAR
                                              Exp (4)
                                                INT: 203
                                          RP
                                        PLUS
                                        Exp (4)
                                          INT: 168
                                      OR
                                      Exp (4)
                                        FLOAT: 8394.000000
                                    OR
                                    Exp (4)
                                      Exp (4)
                                        MINUS
                                        Exp (4)
                                          ID: tcFUA9
                                      RELOP
                                      Exp (4)
                                        LP
                                        Exp (4)
                                          INT: 3
                                        RP
                                  ASSIGNOP
                                  Exp (4)
                                    Exp (4)
                                      INT: 188684
                                    ASSIGNOP
                                    Exp (4)
                                      Exp (4)
                                        INT: 18
                                      STAR
                                      Exp (4)
                                        NOT
                                        Exp (4)
                                          ID: gHnDpr2
                                          LP
                                          Args (4)
                                            Exp (4)
                                              ID: qouWvU9
                                            COMMA
                                            Args (4)
                                              Exp (4)
                                                ID: PLRI
                                              COMMA
                                              Args (4)
                                                Exp (4)
                                                  ID: aJvTKi
                                                  LP
                                                  Args (4)
                                                    Exp (4)
                                                      ID: E0o02ZA
                                                    COMMA
                                                    Args (4)
                                                      Exp (4)
                                                        Exp (4)
                                                          Exp (4)
                                                            Exp (4)
                                                              Exp (4)
                                                                Exp (4)
                                                                  FLOAT: 110539.601562
                                                                PLUS
                                                                Exp (4)
                                                                  Exp (4)
                                                                    Exp (4)
                                                                      ID: iCAPvAM1
                                                                    LB
                                                                    Exp (4)
                                                                      Exp (4)
                                                                        INT: 3491
                                                                      ASSIGNOP
                                                                      Exp (4)
                                                                        FLOAT: 141.163895
                                                                    RB
                                                                  DIV
                                                                  Exp (4)
                                                                    FLOAT: 6.774000
                                                              OR
                                                              Exp (4)
                                                                Exp (4)
                                                                  Exp (4)
                                                                    ID: hxSRnq5
                                                                  STAR
                                                                  Exp (4)
                                                                    INT: 3882
                                                                AND
                                                                Exp (4)
                                                                  Exp (4)
                                                                    Exp (4)
                                                                      Exp (4)
                                                                        Exp (4)
                                                                          Exp (4)
                                                                            INT: 94
                                                                          DIV
                                                                          Exp (4)
                                                                            ID: Sf6ZMNjMM
                                                                        RELOP
                                                                        Exp (4)
                                                                          Exp (4)
                                                                            ID: h67LRgL
                                                                          DIV
                                                                          Exp (4)
                                                                            MINUS
                                                                            Exp (4)
                                                                              INT: 2373
                                                                      RELOP
                                                                      Exp (4)
                                                                        INT: 40300
                                                                    RELOP
                                                                    Exp (4)
                                                                      Exp (4)
                                                                        FLOAT: 823.549988
                                                                      PLUS
                                                                      Exp (4)
                                                                        NOT
                                                                        Exp (4)
                                                                          ID: _raTa
                                                                          LP
                                                                          RP
                                                                  RELOP
                                                                  Exp (4)
                                                                    LP
                                                                    Exp (4)
                                                                      Exp (4)
                                                                        FLOAT: 917.099976
                                                                      MINUS
                                                                      Exp (4)
                                                                        INT: 3061
                                                                    RP
                                                            OR
                                                            Exp (4)
                                                              Exp (4)
                                                                NOT
                                                                Exp (4)
                                                                  MINUS
                                                                  Exp (4)
                                                                    INT: 88
                                                              DIV
                                                              Exp (4)
                                                                ID: iLtmQ2tCy
                                                          OR
                                                          Exp (4)
                                                            Exp (4)
                                                              ID: Laid2RnZ
                                                            DOT
                                                            ID: AhrAlj
                                                        ASSIGNOP
                                                        Exp (4)
                                                          Exp (4)
                                                            LP
                                                            Exp (4)
                                                              ID: OdQPWvGa
                                                            RP
                                                          AND
                                                          Exp (4)
                                                            INT: 4285
                                                      COMMA
                                                      Args (4)
                                                        Exp (4)
                                                          ID: NSyKr
                                                        COMMA
                                                        Args (4)
                                                          Exp (4)
                                                            INT: 155
                                                          COMMA
                                                          Args (4)
                                                            Exp (4)
                                                              ID: ru88E1v
                                                              LP
                                                              RP
                                                  RP
                                                COMMA
                                                Args (4)
                                                  Exp (4)
                                                    Exp (4)
                                                      Exp (4)
                                                        LP
                                                        Exp (4)
                                                          ID: Cbv_Bq8Jc
                                                        RP
                                                      MINUS
                                                      Exp (4)
                                                        NOT
                                                        Exp (4)
                                                          Exp (4)
                                                            ID: CHTt8HaK
                                                          LB
                                                          Exp (4)
                                                            ID: JcC3
                                                            LP
                                                            RP
                                                          RB
                                                    MINUS
                                                    Exp (4)
                                                      LP
                                                      Exp (4)
                                                        Exp (4)
                                                          ID: sXkry7ail
                                                          LP
                                                          RP
                                                        LB
                                                        Exp (4)
                                                          ID: N2Uz0L
                                                        RB
                                                      RP
                                          RP
                    RP
                  RB
                SEMI
        RC
    ExtDefList (6)
      ExtDef (6)
        Specifier (6)
          TYPE: float
        ExtDecList (6)
          VarDec (6)
            ID: qohNMKpB3S
          COMMA
          ExtDecList (6)
            VarDec (6)
              VarDec (6)
                ID: fm0DU_Qh
              LB
              INT: 48341
              RB
            COMMA
            ExtDecList (6)
              VarDec (6)
                VarDec (6)
                  ID: B6PlP0S0
                LB
                INT: 2124
                RB
              COMMA
              ExtDecList (6)
                VarDec (6)
                  ID: jhwWoRt4
                COMMA
                ExtDecList (6)
                  VarDec (6)
                    ID: n2phZCzbp
        SEMI
      ExtDefList (7)
        ExtDef (7)
          Specifier (7)
            StructSpecifier (7)
              STRUCT
              OptTag (7)
                ID: _OxPpQQ
              LC
              DefList (8)
                Def (8)
                  Specifier (8)
                    TYPE: int
                  DecList (8)
                    Dec (8)
                      VarDec (8)
                        ID: zyzBQk
                    COMMA
                    DecList (8)
                      Dec (8)
                        VarDec (8)
                          ID: mqbddYh
                  SEMI
                DefList (9)
                  Def (9)
                    Specifier (9)
                      TYPE: int
                    DecList (9)
                      Dec (9)
                        VarDec (9)
                          ID: FAGTK
                      COMMA
                      DecList (9)
                        Dec (9)
                          VarDec (9)
                            ID: q3X6coKfm
                          ASSIGNOP
                          Exp (9)
                            INT: 63014
                        COMMA
                        DecList (9)
                          Dec (9)
                            VarDec (9)
                              ID: FB1rbf
                            ASSIGNOP
                            Exp (9)
                              FLOAT: 1718279995392.000000
                    SEMI
              RC
          FunDec (11)
            ID: JfaFtGL
            LP
            VarList (11)
              ParamDec (11)
                Specifier (11)
                  TYPE: float
                VarDec (11)
                  ID: HkEQoZPrV
              COMMA
              VarList (11)
                ParamDec (11)
                  Specifier (11)
                    TYPE: int
                  VarDec (11)
                    VarDec (11)
                      ID: CWdOQ
                    LB
                    INT: 45980
                    RB
            RP
          CompSt (11)
            LC
            StmtList (12)
              Stmt (12)
                WHILE
                LP
                Exp (12)
                  INT: 211
                RP
                Stmt (12)
                  IF
                  LP
                  Exp (12)
                    Exp (12)
                      FLOAT: 5.924000
                    LB
                    Exp (12)
                      ID: htNKoAV
                    RB
                  RP
                  Stmt (12)
                    Exp (12)
                      FLOAT: 902.150024
                    SEMI
              StmtList (13)
                Stmt (13)
                  Exp (13)
                    Exp (13)
                      ID: NTe0jU6
                    OR
                    Exp (13)
                      Exp (13)
                        ID: UUXOBoap2y
                      MINUS
                      Exp (13)
                        Exp (13)
                          FLOAT: 0.018753
                        DOT
                        ID: uqBo7A_
                  SEMI
            RC
        ExtDefList (15)
          ExtDef (15)
            Specifier (15)
              StructSpecifier (15)
                STRUCT
                Tag (15)
                  ID: FNtTI_Kfnh
            SEMI
          ExtDefList (16)
            ExtDef (16)
              Specifier (16)
                TYPE: int
              ExtDecList (16)
                VarDec (16)
                  ID: iK1NbTFcE
              SEMI
            ExtDefList (17)
              ExtDef (17)
                Specifier (17)
                  TYPE: int
                ExtDecList (17)
                  VarDec (17)
                    VarDec (17)
                      ID: GN13DP
                    LB
                    INT: 9
                    RB
                SEMI
